simpleocr/files.py

```diff
@@ -45,45 +32,152 @@ def write(self):
+                # If the file is not available, set ground data to None
+                self.ground = None
         else:
-            self.ground_path = basepath + GROUND_EXTENSIONS_DEFAULT
+            # No file reference is available, so no ground data is available as the file has not yet been grounded
             self.ground = None
+            self.ground_path = None
+
+    @staticmethod
+    def from_file(path):
+        """
+        Create an image object with a path to an image. As the path is
+        put through the get_absolute_path function, the valid inputs are the same
+        as for the path passed to that function
+        :param path: image path, str
+        :return: Image object with file reference
+        """
+        path = Image.get_absolute_path(path)
+        array = cv2.imread(path)
+        return Image(array=array, path=path)
+
+    @staticmethod
+    def from_pil(pillow):
+        """ Create an Image object based off of a Pillow Image """
+        return Image(Image.get_array_from_pil(pillow))
+
+    @staticmethod
+    def get_absolute_path(image):
+        """
+        Get the absolute path for an image. Valid inputs are:
+        - Relative path with or without extension
+        - File in DATA_DIRECTORY with or without extension
+        - Absolute path with or without extension
+        :param image: image path in str type
+        :return: The absolute image path
+        """
+        # If the path exists, return the path, but make sure it's an absolute path first
+        if os.path.exists(image):
+            return os.path.abspath(image)
+        # Try to find the file with the passed path with the various extensions
+        image_with_extension = try_extensions(IMAGE_EXTENSIONS, os.path.splitext(image)[0])
+        if image_with_extension:
+            return os.path.abspath(image_with_extension)
+        # The file must be in the data directory if it has not yet been found
+        image_basename = os.path.basename(image)
+        image_datadir = try_extensions(IMAGE_EXTENSIONS, os.path.join(DATA_DIRECTORY, image_basename))
+        if image_datadir:
+            return os.path.abspath(image_datadir)
+        # The file cannot be found, so raise a FileNotFound Error
+        raise Image.FileNotFound
+
+    @staticmethod
+    def get_array_from_pil(pillow):
+        """
+        Returns an OpenCV compatible array from a Pillow Image object
+        :param pillow: pillow image object
+        :return: OpenCV compatible numpy array
+        """
+        imagefile = numpy.array(pillow)
+        return imagefile[:, :, ::-1].copy()
 
     @property
     def is_grounded(self):
-        """checks if this file is grounded"""
+        """
+        :return: True if Image is grounded
+        """
         return not (self.ground is None)
 
-    def set_ground(self, segments, classes, write_file=False):
-        """creates the ground, saves it to a file"""
-        if self.is_grounded:
+    @property
```

Comment:
I don't think there's a need to have this, `image.path is None` is quite readable